### PR TITLE
BREAKING(package): update SUI to 2.3

### LINKED ISSUE #2550: Update SUIR to be compatible with SUI 2.3
Creating this as a container for referencing other issues that are likely to pop up with the release of SUI 2.3 which happened today. Feel free to edit this and add related issues below for tracking. 

Notes about what is new in this release:
https://semantic-ui.com/introduction/new.html

Full CHANGELOG:
https://github.com/Semantic-Org/Semantic-UI/blob/master/RELEASE-NOTES.md

Known issues so far. Please edit this list to add or link to issues:
- [x] Fix modal placement (#2549) 
- [x] Update `Icon` component to use new FontAwesome 5.0 icons.
- [x] Add support for new Transition names `zoom` and `glow`
- [x]  Modal support to disable vertical centering ([missing from the above linked notes. See examples.](https://semantic-ui.com/modules/modal.html#/examples))
- [ ] Update Search logic for `fullTextSearch: 'exact'` vs `fuzzy`

### PR DESCRIPTION
**Pull request by layershifter**
# BREAKING

### This PR requieres usage of Semantic UI CSS >= 2.3.x

----

Fixes #2550. This PR should:

- [x] update `Dimmer` to be compatible with SUI 2.3 (#2656)
- [x] update `Modal` to be compatible with SUI 2.3 (#2690)
- [x] update `Icon` to use FA5 (#2665)
- [x] add support for new `Transition` names: `zoom` and `glow` (#2658)

### CODE REVIEW

**Comment by levithomason** [comment]
We should update the install docs to note SUI >=2.3 is required.

**Comment by ghost** [comment]
There is a message for Modals in 2.3.1: https://github.com/Semantic-Org/Semantic-UI/blob/master/RELEASE-NOTES.md.

> A Special Message about Flex Modals There will be an update shortly to resolve issues related to flex modals when using multiple modals and detachable: false, in order to not hold up this release, we've decided to move forward without a fix.
> 
> A general solution will most likely require branching code for IE11 which will disable flex (as IE11 doesnt correctly implement the latest spec for absolute positioned flex containers).

**Comment by chuckhagy** [comment]
This PR fixes an app I'm working on... can I get an ETA on when it's expected to be merged in?

**Comment by layershifter** [comment]
This PR requires some chore work before merge, I want to finish it on this weekend.

**Comment by brianespinosa** [comment]
It doesn't look like there is a change made to the README.md file to remove the warning at the very top about incompatibility with SUI =< 2.3. We should probably remove that.

> We should update the install docs to note SUI >=2.3 is required.

We should definitely have something prominently displayed still regarding the above note.

**Comment by layershifter** [comment]
Changes that were made unded Semantic-Org/Semantic-UI#6226 made icon generator useless.

I'll manually generate icons and add notice about versions.

File: README.md [hunk 1]
```diff
@@ -1,5 +1,3 @@
-⚠️NOTE: Currently Semantic UI React is not yet compatible with the latest 2.3 version of Semantic UI styles. See [#2250](https://github.com/Semantic-Org/Semantic-UI-React/issues/2550) for more info.
-
```
Comment (layershifter) [inline README.md:2]: Now we can remove this note :+1:

File: docs/app/Views/Usage.js [hunk 2]
```diff
@@ -66,7 +57,21 @@ const Usage = () => (
...
+        <Message.Header as='h3'>Semantic UI support</Message.Header>
+        <p>
+          The release 2.3 of Semantic UI introduced some backward incompatible changes, so you should use corresponding
+          version of Semantic UI React:
+        </p>
+        <Message.List>
+          <Message.Item>for SUI 2.2 use 0.79.1 and below</Message.Item>
+          <Message.Item>for SUI 2.3 use 0.80.0 and higher</Message.Item>
+        </Message.List>
+      </Message>
```
Comment (layershifter) [inline docs/app/Views/Usage.js:74]: Notice about version support

**Comment by layershifter** [comment]
Marked as ready for review :+1:

**Review by brianespinosa — APPROVED** [review]

**Comment by levithomason** [comment]
Hope to add my review by EOD Monday.

**Comment by aminland** [comment]
Is there an ETA on when this might be merged?

**Comment by layershifter** [comment]
I merged this branch with `master`, waiting for @levithomason

**Review by levithomason — APPROVED** [review]

**Comment by levithomason** [comment]
Released in `semantic-ui-react@0.81.0`.

**Pull request merged**